File: Minimize the Heights II/sol.java
class Solution {
    int getMinDiff(int[] arr, int n, int k) {
        Arrays.sort(arr);
        int min = arr[0];
        int max = arr[n-1];
        int diff = max - min;
        
        for(int i=1; i<n; i++){
            if(arr[i]-k <= 0){
                continue;
            }
            
            min = Math.min(arr[0] + k, arr[i] - k);
            max = Math.max(arr[n-1]-k, arr[i-1]+k);
            diff = Math.min(diff, max-min);
        }
        
        return diff;
    }
}

File: Merge Without Extra Space/sol.py
class Solution:
    
    #Function to merge the arrays.
    def merge(self,arr1,arr2,n,m):
        i = n - 1
        j = 0

        while i >= 0 and j < m:
            if arr1[i] > arr2[j]:
                temp = arr1[i]
                arr1[i] = arr2[j]
                arr2[j] = temp
                i -= 1
                j += 1
            else:
                break

        arr1.sort()
        arr2.sort()
